libr/core/cmd_print.c

```diff
@@ -3333,6 +3333,73 @@ static bool cmd_print_blocks(RCore *core, const char *input) {
 	return result;
 }
 
+static bool cmd_pse(RCore *core, const char *input) {
+	if (input[2] == ' ') {
+		const char *fname = r_str_trim_head_ro (input + 3);
+		char *tmp = r_str_newf (R_JOIN_2_PATHS (R_JOIN_2_PATHS (R2_HOME_DATADIR, "util"), "%s"), fname);//todo: add R2_HOME_UTIL
+		char *home = r_str_home (tmp);
+		free (tmp);
+		tmp = r_str_newf (R_JOIN_2_PATHS (R_JOIN_2_PATHS (R2_SDB, "util"), "%s"), r_str_newf (R_JOIN_2_PATHS ("d", "%s"), fname));
+		char *path = r_str_r2_prefix (tmp);
+
+		if (!sdb_query_file (core->anal->charset_db, path)) {
+			eprintf ("Path: %s not found\n", path);
+		}
+		free (home);
+		free (path);
+		free (tmp);
+
+
+		SdbListIter *iter;
+		SdbKv *kv;
+		SdbList *sdbls = sdb_foreach_list (core->anal->charset_db, true);
+		size_t i = 0;
+
+
+		core->custom = r_charset_new ();
+		ls_foreach (sdbls, iter, kv) {
+			ut8 *buff_hex = (ut8 *) sdbkv_key (kv);
+			ut8 *buff_char = (ut8 *) sdbkv_value (kv);
+
+			ut64 mylonglong = r_num_get (NULL, (char *)buff_hex);
+			ut8 *hex = (ut8 *) malloc (sizeof (ut64));
+			memcpy (hex, (ut8 *)&mylonglong, sizeof (ut64));
+			hex[sizeof (hex)] = '\0';
```

Comment:
uhm... why null terminate a binary buffer? i dont get that. also this is affected by the endianness. you may want to use r_read_le apis

Reply:
with not null terminating string you have `Selfreplicating----arbitrary-code------by-MrCheeze-:)---------------------\\AvvJOHN\\]V `instead of `Selfreplicating----arbitrary-code------by-MrCheeze-:)---------------------`

Reply:
Ah no wrong \0! this one is useless!